### PR TITLE
feat: replica exec send reply to client

### PR DESCRIPTION
**Pull request by liubaohai**
# Description           

# Checklist:

- [x] **Self-review**: I have performed a **self-review** of my own code
- [ ] **Comment**:     I have **commented my code**, particularly in hard-to-understand areas
- [ ] **Doc**:         I have made corresponding changes to the **documentation**
- [x] **No-warnings**: My changes generate **no new warnings**
- [x] **Add-test**:    I have added **tests** that prove my fix is effective or that my feature works
- [x] **Pass**:        New and existing **unit tests pass** locally with my changes

### CODE REVIEW

File: components/epaxos/src/replica/exec.rs [hunk 1]
```diff
@@ -70,14 +66,32 @@ impl Replica {
         None
     }
 
-    pub fn execute_commands(
+    async fn send_reply(&self, mut replys: HashMap<InstanceId, Vec<ExecuteResult>>) {
+        let mut txes = self.reply_txes.lock().await;
+        for (k, v) in replys.drain() {
```
Comment (drmingdrmer) [inline components/epaxos/src/replica/exec.rs:71]: 这么看来replys不许hashMap, `Vec<(InstanceId, Vec<ExecuteResult>>`就够了

File: components/epaxos/src/replica/exec.rs [hunk 2]
```diff
@@ -70,14 +66,32 @@ impl Replica {
...
+            let tx = match txes.remove(&k) {
+                Some(t) => t,
+                None => continue,
+            };
+            if let Err(_) = tx.send(v) {
+                println!("the receiver dropped for {:?}", k);
+            }
```
Comment (drmingdrmer) [inline components/epaxos/src/replica/exec.rs:78]: 加个dropped rx 测试吧,保证这块能正常运行过去不panic啥的

File: components/epaxos/src/replica/replica.rs [hunk 3]
```diff
@@ -78,13 +81,20 @@ impl From<(ReplicaId, String, bool)> for ReplicaPeer {
...
+}
+
 /// structure to represent a replica
 pub struct Replica {
     pub replica_id: ReplicaId,
     pub group_replica_ids: Vec<ReplicaId>,
     pub peers: Vec<ReplicaPeer>,
     pub storage: Storage,
     pub committed_timeout: i32,
+    pub reply_txes: Mutex<HashMap<InstanceId, Sender<Vec<ExecuteResult>>>>,
```
Comment (drmingdrmer) [inline components/epaxos/src/replica/replica.rs:97]: 换这个名字咋样? `waiting_replies`, tx的意思应该能从类型看出来了, 名字感觉主要用来表示业务上的含义好点.
```suggestion
    pub waiting_replies: Mutex<HashMap<InstanceId, Sender<Vec<ExecuteResult>>>>,
```

File: components/epaxos/src/replica/exec.rs (hunk 1 above)
Comment (liubaohai) [reply components/epaxos/src/replica/exec.rs:71]: 嗯，用vec好点

File: components/epaxos/src/replica/exec.rs (hunk 2 above)
Comment (liubaohai) [reply components/epaxos/src/replica/exec.rs:78]: 好的

**Pull request merged**